src/Microsoft.AspNetCore.Razor.Tools/ServerProtocol/ServerConnection.cs

```diff
@@ -21,6 +21,9 @@ internal static class ServerConnection
         // Spend up to 20s connecting to a new process, to allow time for it to start.
         private const int TimeOutMsNewProcess = 20000;
 
+        // Custom delegate that contains an out param to use with TryCreateServerCore method.
+        private delegate TResult Func<T1, T2, T3, T4, out TResult>(T1 arg1, T2 arg2, out T3 arg3, T4 arg4);
```

Comment:
Sorry, what's happening here?

Reply:
Needed a custom delegate to support the out parameter. The built-in Func<> overloads wouldn't work.

Reply:
Ok. Let's give this a different name `TryCreateServerDelegate` and this should be good to go.